src/auth/digest/Config.cc

```diff
@@ -807,8 +808,14 @@ Auth::Digest::Config::decode(char const *proxy_auth, const char *aRequestRealm)
         switch (t) {
         case DIGEST_USERNAME:
             safe_free(username);
-            if (value.size() != 0)
+            if (value.size() != 0) {
+                const auto v = value.rawBuf();
+                if (utf8 && !isValidUtf8String(v, v + value.size())) {
+                    auto str = isCP1251EncodingAllowed(request) ? Cp1251ToUtf8(v) : Latin1ToUtf8(v);
+                    value.limitInit(str.rawContent(), str.length());
```

Comment:
Jenkins tests are currently failing on this line with:  error: 'class String' has no member named 'limitInit'

```suggestion
                    value = SBufToString(str);
```